src/library/basesqltablemodel.cpp

```diff
@@ -293,7 +324,7 @@ void BaseSqlTableModel::select() {
     // should not disturb that if we are only removing tracks.
     qStableSort(rowInfo.begin(), rowInfo.end());
 
-    m_trackIdToRows.clear();
+    TrackId2Rows trackIdToRows;
```

Comment:
Ok, m_trackIdToRows is cleared and and re-populated whereas m_rowInfo is cleared but then replaced by a new vector. But why this nonuniform handling? Some comments that this is an optimization and not arbitrary would have been helpful ;)

I will compare the timings.